### PR TITLE
Cortx-31170: CSM user management and IAM users management API

### PR DESCRIPTION
**Pull request by abhijit1patil**
# Problem Statement
- [CORTX-31170 JIRA ticket](https://jts.seagate.com/browse/CORTX-31170)

# Design
-  [Management API Changes](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1006305559/Management+API+Changes)

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: config/csm/csm_config.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ Restcall:
   s3accounts_endpoint : "/api/v2/s3_accounts"
   s3_bucket_endpoint : "/api/v2/s3/bucket"
   IAM_users_endpoint : "/api/v2/iam_users"
-  s3_iam_user_endpoint : "/api/v2/s3/iam/users"
-  csmuser_endpoint : "/api/v2/csm/users"
+  s3_iam_user_endpoint : "/api/v2/iam/users"
+  csmuser_endpoint : "/api/v2/system/users"
```
Comment (kachhwahadivya) [inline config/csm/csm_config.yaml:29]: This is for LR so we can skip the change from here

File: config/csm/csm_config.yaml [hunk 2]
```diff
@@ -84,11 +84,11 @@ Restcall_LC:
...
+  s3_iam_user_endpoint : "/api/v2/iam/users"
   s3_iam_keys_endpoint : "/api/v2/s3/iam/keys"
-  s3_iam_caps_endpoint : "/api/v2/s3/iam/caps"
+  s3_iam_caps_endpoint : "/api/v2/iam/caps"
   get_set_quota: "/api/v2/iam/quota/{}"
-  csmuser_endpoint : "/api/v2/csm/users"
+  csmuser_endpoint : "/api/v2/system/users"
   csmuser_permission_endpoint: "/api/v2/permissions"
   bucket_policy_endpoint : "/api/v2/s3/bucket_policy/{}"
   perf_stats_endpoint: "/api/v2/metrics/perf"
```
Comment (kachhwahadivya) [inline config/csm/csm_config.yaml:94]: degraded_cap_endpoint: "/api/v2/capacity/status" this needs to be changed to degraded_cap_endpoint: "/api/v2/capacity/status"

File: config/csm/csm_config.yaml (hunk 1 above)
Comment (abhijit1patil) [reply config/csm/csm_config.yaml]: updated

**Review by kachhwahadivya — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
Approving for merge.

**Pull request merged**